### ISSUE TITLE
'dtype' does not refer to a value with C++11

### ISSUE THREAD

**Issue by ankane** [dpilger26/NumCpp#50]
**Describe the bug**

There's a compilation error with C++11. The error occurs in both basic programs as well as the examples.

```txt
In file included from ReadMe.cpp:1:
In file included from /usr/local/include/NumCpp.hpp:34:
In file included from /usr/local/include/NumCpp/Coordinates.hpp:31:
In file included from /usr/local/include/NumCpp/Coordinates/Coordinate.hpp:31:
In file included from /usr/local/include/NumCpp/Coordinates/Dec.hpp:33:
In file included from /usr/local/include/NumCpp/Functions/deg2rad.hpp:31:
In file included from /usr/local/include/NumCpp/NdArray.hpp:32:
/usr/local/include/NumCpp/NdArray/NdArrayOperators.hpp:805:47: error: 'dtype' does not refer to a value
            returnArray.begin(), std::bit_not<dtype>());
                                              ^
/usr/local/include/NumCpp/NdArray/NdArrayOperators.hpp:797:23: note: declared here
    template<typename dtype>
                      ^
/usr/local/include/NumCpp/NdArray/NdArrayOperators.hpp:805:39: error: no member named 'bit_not' in namespace 'std'
            returnArray.begin(), std::bit_not<dtype>());
                                 ~~~~~^
/usr/local/include/NumCpp/NdArray/NdArrayOperators.hpp:805:54: error: expected expression
            returnArray.begin(), std::bit_not<dtype>());
                                                     ^
ReadMe.cpp:89:10: warning: decomposition declarations are a C++17 extension [-Wc++17-extensions]
    auto [rows, cols] = nc::nonzero(a);
         ^~~~~~~~~~~~
1 warning and 3 errors generated.
```

**To Reproduce**

```sh
cd examples/ReadMe
c++ -std=c++11 ReadMe.cpp -o test -lboost_filesystem
```

**Expected behavior**

The program compiles, as it does with C++14 and C++17.

**Compiler Info**

```sh
c++ --version
Apple LLVM version 10.0.1 (clang-1001.0.46.4)
Target: x86_64-apple-darwin18.7.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

**Comment by dpilger26** [comment]
It looks like ```std::bit_not``` wasn't added to the standard until C++14.  Oddly enough my versions of GNU and Clang compilers both happily compiled ```std::bit_not``` in C++11 mode.  None the less,  I've removed that function and replaced with my own so NumCpp _should_ (let me know if you run into any other errors...) be back to fully C++11 compatible.  I also added a ```__cpp_structured_bindings``` check around the C++17 structured bindings in the ReadMe example code to get rid of that compiler warning.
